src/tests/large-mpi.cpp

```diff
@@ -0,0 +1,128 @@
+    /* Load RSA pubkey of PGP_MPINT_BITS size (16384). Must succeed. */
+    assert_rnp_success(
+      rnp_input_from_path(&input, "data/test_large_MPIs/rsa-pub-16384bits.pgp"));
+    assert_rnp_success(rnp_import_keys(ffi, input, RNP_LOAD_SAVE_PUBLIC_KEYS, NULL));
+    rnp_input_destroy(input);
+
+    /* Load RSA signature. rsa-pub-65535bits.pgp file signed by previously loaded 16384 bit RSA
+     * key. Must succeed. */
+    assert_rnp_success(
+      rnp_input_from_path(&input, "data/test_large_MPIs/rsa-pub-65535bits.pgp"));
+    assert_rnp_success(
+      rnp_input_from_path(&signature, "data/test_large_MPIs/rsa-pub-65535bits.pgp.sig"));
+    assert_rnp_success(rnp_op_verify_detached_create(&verify, ffi, input, signature));
+    assert_rnp_success(rnp_op_verify_execute(verify));
+    rnp_input_destroy(input);
+    rnp_input_destroy(signature);
+    assert_rnp_success(rnp_op_verify_destroy(verify));
+
+    /* Load RSA signature with PGP_MPINT_BITS + 1 size (16385) MPI. Must fail. */
+    assert_rnp_success(
+      rnp_input_from_path(&input, "data/test_large_MPIs/rsa-pub-65535bits.pgp"));
+    assert_rnp_success(rnp_input_from_path(
+      &signature, "data/test_large_MPIs/rsa-pub-65535bits.pgp.16385sig.sig"));
+    assert_rnp_success(rnp_op_verify_detached_create(&verify, ffi, input, signature));
+    assert_rnp_failure(rnp_op_verify_execute(verify));
+    rnp_input_destroy(input);
+    rnp_input_destroy(signature);
+    assert_rnp_success(rnp_op_verify_destroy(verify));
+
+    rnp_ffi_destroy(ffi);
+}
+
+TEST_F(rnp_tests, test_large_mpi_rsa_priv)
+{
+    rnp_ffi_t    ffi = NULL;
+    rnp_input_t  input = NULL;
+    rnp_output_t output = NULL;
+
+    assert_rnp_success(rnp_ffi_create(&ffi, "GPG", "GPG"));
+
+    /* Load RSA private key of PGP_MPINT_BITS + 1 size (16385). Must fail. */
+    assert_rnp_success(
+      rnp_input_from_path(&input, "data/test_large_MPIs/rsa-priv-16385bits.pgp"));
+    assert_rnp_failure(rnp_import_keys(ffi, input, RNP_LOAD_SAVE_SECRET_KEYS, NULL));
+    rnp_input_destroy(input);
+
+    /* Load RSA private key of PGP_MPINT_BITS size (16384). Must succeed. */
+    assert_rnp_success(
+      rnp_input_from_path(&input, "data/test_large_MPIs/rsa-priv-16384bits.pgp"));
+    assert_rnp_success(rnp_import_keys(ffi, input, RNP_LOAD_SAVE_SECRET_KEYS, NULL));
+    rnp_input_destroy(input);
+
+    /* Load PKESK encrypted by PGP_MPINT_BITS sized key. Must succeed. */
+    assert_rnp_success(
+      rnp_input_from_path(&input, "data/test_large_MPIs/message.enc.rsa16384.pgp"));
+    assert_rnp_success(rnp_output_to_null(&output));
+    assert_rnp_success(rnp_decrypt(ffi, input, output));
+    rnp_input_destroy(input);
+    rnp_output_destroy(output);
+
+    /* Load PKESK having 16385 bit MPI. Must fail. */
+    assert_rnp_success(
+      rnp_input_from_path(&input, "data/test_large_MPIs/message.enc.rsa16385.pgp"));
+    assert_rnp_success(rnp_output_to_null(&output));
+    assert_rnp_failure(rnp_decrypt(ffi, input, output));
```

Comment:
I would test against a specific error code to make sure we're really getting the MPI parse error:
`assert_int_equal(rnp_decrypt(ffi, input, output), RNP_ERROR_BAD_FORMAT);`